aten/src/ATen/native/Sorting.cpp

```diff
@@ -272,6 +268,148 @@ std::tuple<Tensor&, Tensor&> kthvalue_out_impl_cpu(
   return std::forward_as_tuple(values, indices);
 }
 
+// Computes both the median and its index along dimension dim of the input
+std::tuple<Tensor&, Tensor&> median_with_indices_impl(
+    Tensor& values,
+    Tensor& indices,
+    const Tensor& self,
+    int64_t dim,
+    bool keepdim,
+    bool ignore_nan) {
+  NoNamesGuard guard;
+  dim = at::maybe_wrap_dim(dim, self.dim());
+
+  int64_t size = self.dim() > 0 ? self.size(dim) : 1;
+  TORCH_CHECK(
+      size > 0,
```

Comment:
Should this also check for the size of the specified dimension being zero?

Reply:
That is what's doing but I'll fix the error messaging which is misleading.